framework/tests/flow2/Purity.java

```diff
@@ -182,6 +186,26 @@ void nonpure() {
         return "";
     }
 
+    @Deterministic String t17a(Purity l) {
+      //:: error: (purity.not.deterministic.assign.field)
+      f1 = "";
```

Comment:
Is this a false positive? It seems pretty deterministic behavior - field f1 is always updated with a literal.
Can this be prevented by using the Constant Value Checker and a simple CFG analysis?